File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
let idx = 0;
const [N, M] = input[idx++].split(' ').map(Number);
const graph = Array.from({length: N + 1}, () => []);
const visited = new Array(N + 1).fill(false);

for (let i = 0; i < M; i++) {
    const [a, b] = input[idx++].split(' ').map(Number);
    graph[a].push(b);
    graph[b].push(a);
}

function dfs(start) {
    const stack = [start];
    visited[start] = true;
    while (stack.length > 0) {
        const v = stack.pop();
        for (let i = graph[v].length - 1; i >= 0; i--) {
            const next = graph[v][i];
            if (!visited[next]) {
                visited[next] = true;
                stack.push(next);
            }
        }
    }
}

let count = 0;
for (let i = 1; i <= N; i++) {
    if (!visited[i]) {
        dfs(i);
        count++;
    }
}
console.log(count);

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
import sys
sys.setrecursionlimit(10000)
input = sys.stdin.readline

N, M = map(int, input().split())
graph = [[] for _ in range(N + 1)]
visited = [False] * (N + 1)

def dfs(v):
    visited[v] = True
    for next_v in graph[v]:
        if not visited[next_v]:
            dfs(next_v)

for _ in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
    
count = 0
for i in range(1, N + 1):
    if not visited[i]:
        dfs(i)
        count += 1

print(count)